### ISSUE TITLE
strange behavior of current_setting(app.settings.*)

### ISSUE THREAD

**Issue by sunbing81** [PostgREST/postgrest#1141]
### Environment

* PostgreSQL version: 10.3
* PostgREST version: 0.5
* Operating system: AWS RDS

### Description of issue

The case is simple. Let the code talk.

```sql
create or replace function api.test() returns text AS $$
select current_setting('app.settings.test');
$$ language sql immutable strict;
```

Of course we have the corresponding setting in postgrest.conf.
```
app.settings.test = "a_test"
```

Immediately after a fresh start of postgrest, a correct response is produced by
```
GET /rpc/test
```

The response will be always correct as long as the request is repeated ***soon enough***. Now it comes the problem. If the request is issued a certain amount of time after a fresh start of postgrest, or a correct response, we would have the exception of 'unrecognized configuration parameter \"app.settings.test\"'.

Is this possibly a bug, or am I missing anything?

**Comment by sunbing81** [comment]
Alright, it seems that all the session settings are affected, including request.jwt.claim.* and others. This is really a bizarre problem since everything used to work well before.

**Comment by steve-chavez** [comment]
Reproduced. Seems this is related to the pool timeout of [10 seconds](https://github.com/PostgREST/postgrest/blob/master/main/Main.hs#L172)(I increased this value to confirm), these `app.settings.*` GUCs are session scoped and set up at startup, when the pool reaper kills connections the new ones don't get these GUCs.

Seems this was also noted in https://github.com/PostgREST/postgrest/pull/1063#issuecomment-364670657, though our test suite didn't detect this. Simplest solution that occurs to me is to make these settings transaction scoped. 

@canadaduane Have you runned into this issue, if so how are you handling it?
